components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4591,6 +4614,10 @@ where
                 .unwrap();
 
             if !campaigned {
+                // The new peer has not campaigned yet, record it for later campaign.
+                if !is_leader {
```

Comment:
```suggestion
                if !is_leader && self.fsm.peer.raft_group.raft.state == StateRole::Follower {
```
No need to check of self is not voter.